operator/v1/types_etcd.go

```diff
@@ -26,6 +26,18 @@ type EtcdSpec struct {
 
 type EtcdStatus struct {
 	StaticPodOperatorStatus `json:",inline"`
+
+	// leader describes the etcd member which is currently the leader. This field
+	// is a hint and shouldn't be considered authoritative.
+	Leader LeaderStatus `json:"leader"`
+}
+
+// leaderStatus describes the etcd leader member details.
+type LeaderStatus struct {
+	// name is the etcd leader member name, if available.
+	Name string `json:"name,omitempty"`
+	// node is the etcd leader member node, if available.
+	Node string `json:"node,omitempty"`
```

Comment:
Shouldn't this be e.g. `NodeRef corev1.ObjectReference` ?

Reply:
> Shouldn't this be e.g. `NodeRef *corev1.ObjectReference` ?

No.  We encourage the creation of specific reference types.  See the reasoning in https://github.com/kubernetes/api/blob/master/core/v1/types.go#L5172-L5186

@cgwalters

Reply:
Ah, I see.  Hmm.  So, since there's nothing more we care about here to a node than its name, does that argue for keeping it as a `string`?